libcloudforensics/aws.py

```diff
@@ -0,0 +1,694 @@
+        else:
+          response = client.describe_instances(Filters=filters)
+      except client.exceptions.ClientError as exception:
+        raise RuntimeError('Could not retrieve instances: {0:s}'.format(
+            str(exception)))
+
+      for reservation in response['Reservations']:
+        for instance in reservation['Instances']:
+          # If reservation['Instances'] contains any entry, then the
+          # instance's state is expected to be present in the API's response.
+          if instance['State']['Name'] == 'terminated' and not show_terminated:
+            continue
+
+          zone = instance['Placement']['AvailabilityZone']
+          instance_info = {
+              'region': zone[:-1],
+              'zone': zone
+          }
+
+          for tag in instance.get('Tags', []):
+            if tag.get('Key') == 'Name':
+              instance_info['name'] = tag.get('Value')
+              break
+
+          instances[instance['InstanceId']] = instance_info
+
+      next_token = response.get('NextToken')
+      if not next_token:
+        break
+
+    return instances
+
+  def ListVolumes(self, region=None, filters=None):
+    """List volumes of an AWS account.
+
+    Example usage:
+      # List volumes attached to the instance 'some-instance-id'
+      ListVolumes(filters=[
+          {'Name':'attachment.instance-id', 'Values':['some-instance-id']}])
+
+    Args:
+      region (str): Optional. The region from which to list the volumes.
+      filters (list(dict)): Optional. Filter for the query.
+
+    Returns:
+      dict: Dictionary with name and metadata for each volume.
+
+    Raises:
+      RuntimeError: If volumes can't be listed.
+    """
+    if not filters:
+      filters = []
+
+    volumes = {}
+    next_token = None
+    client = self.ClientApi(EC2_SERVICE, region=region)
+
+    while True:
+      try:
+        if next_token:
+          response = client.describe_volumes(
+              Filters=filters, NextToken=next_token)
+        else:
+          response = client.describe_volumes(Filters=filters)
+      except client.exceptions.ClientError as exception:
+        raise RuntimeError('Could not retrieve volumes: {0:s}'.format(
+            str(exception)))
+
+      for volume in response['Volumes']:
+        volume_info = {
+            'region': self.default_region,
+            'zone': volume['AvailabilityZone']
+        }
+
+        for tag in volume.get('Tags', []):
+          if tag.get('Key') == 'Name':
+            volume_info['name'] = tag.get('Value')
+            break
+
+        if len(volume['Attachments']) > 0:
+          volume_info['device'] = volume['Attachments'][0]['Device']
+
+        volumes[volume['VolumeId']] = volume_info
+
+      next_token = response.get('NextToken')
+      if not next_token:
+        break
+
+    return volumes
+
+  def GetInstancesByNameOrId(self, instance_name=None,
+                             instance_id=None, region=None):
```

Comment:
Please use one parameter per line (or all on the same line when possible) for better readability